lib/collection/proxy-config.js

```diff
@@ -0,0 +1,230 @@
+        *    processing.
+        * 2. Checks whether the matchPattern follows the rules, https://developer.chrome.com/extensions/match_patterns,
+        *    If not then, dont process it. It is a Regex match process which is slower, @todo needs to be cached in
+        *    future.
+        * 3. Check for the protocol, as it is a normal array check.
+        * 4. Checks the host, as it doesn't involve only string comparisons.
+        * 5. Finally, checks for the path, which actually involves the Regex matching, the slow process.
+        */
+        var matchRegexObject = {},
+            match = null,
+            url,
+            hostSuffix = '';
+
+        // If the matchPattern is <all_urls> then there is no need for any validations.
+        if (this.match === MATCH_ALL_URLS) {
+            return true;
+        }
+
+        // Check the match pattern of sanity and split it into protocol, host and path
+        match = this.match.match(regexes.patternSplit);
+
+        if (!match) {
+            // This ensures it is a invalid match pattern
+            return false;
+        }
+
+        // If the protocol retuns '*' we are going to assign ['https', 'http'] array to it.
+        matchRegexObject.protocol = match[1] === MATCH_ALL ? ALLOWED_PROTOCOLS : [match[1]];
+
+        matchRegexObject.host = match[2];
+
+        if (match[3]) {
+            matchRegexObject.path = this.globPatternToRegexp(match[3]);
+        }
+
+        // We reached here means, the match pattern provided is valid and been parsed to protocol, host and path.
+        // Convert the url provided into a url object, happens only if the match string is parsed properly.
+        try {
+            url = new Url(urlStr);
+        }
+        catch (e) {
+            return false;
+        }
+
+        // Check for the protocol availability in the parsed match protocol array
+        if (!_.includes(matchRegexObject.protocol, url.protocol)) {
+            return false;
+        }
+
+        /*
+        * For Host match, we are considering the port with the host, hence we are using getRemote() instead of getHost()
+        * We need to address three cases for the host urlStr
+        * 1. * It matches all the host + protocol,  hence we are not having any parsing logic for it.
+        * 2. .*foo.bar.com Here the prefix could be anything but it should end with foo.bar.com
+        * 3. foo.bar.com This is the absolute matching needs to done.
+        */
+        if (matchRegexObject.host === MATCH_ALL) {
+          // Don't check anything in host. simple go ahead for the path matching logic
+        }
+        else if (matchRegexObject.host[0] === MATCH_ALL) {
+            // It must be *.foo. We also need to allow foo by itself.
+            hostSuffix = matchRegexObject.host.substr(2);
+            if (url.getRemote() !== hostSuffix && !url.getRemote().endsWith('.' + hostSuffix)) {
```

Comment:
Since you are returning in this block, you should club it together into a single `if`